samples/MvcSample.Web/Startup.cs

```diff
@@ -16,15 +16,75 @@
 #if DNX451
 using Microsoft.Framework.DependencyInjection.Autofac;
 using Microsoft.Framework.Runtime;
+using Microsoft.Framework.Runtime.Infrastructure;
 #endif
+using System.ComponentModel;
 using MvcSample.Web.Filters;
 using MvcSample.Web.Services;
 
-
 namespace MvcSample.Web
 {
     public class Startup
     {
+        public IServiceProvider ConfigureServices(IServiceCollection services)
+        {
+            services.AddCaching();
+            services.AddSession();
+
+            services.AddMvc();
+            services.AddSingleton<PassThroughAttribute>();
+            services.AddSingleton<UserNameService>();
+            services.AddTransient<ITestService, TestService>();
+
+            services.ConfigureMvc(options =>
+            {
+                options.Filters.Add(typeof(PassThroughAttribute), order: 17);
+                options.AddXmlDataContractSerializerFormatter();
+                options.Filters.Add(new FormatFilterAttribute());
+            });
+
+#if DNX451
+            // REVIEW: This is hella hacky
+            var applicationEnvironment = services.BuildServiceProvider().GetRequiredService<IApplicationEnvironment>();
+            var configurationPath = Path.Combine(applicationEnvironment.ApplicationBasePath, "config.json");
+
+            // Set up configuration sources.
+            var configuration = new Configuration()
+                .AddJsonFile(configurationPath)
+                .AddEnvironmentVariables();
+            string diSystem;
+            if (configuration.TryGet("DependencyInjection", out diSystem) &&
+                diSystem.Equals("AutoFac", StringComparison.OrdinalIgnoreCase))
```

Comment:
can the code from line 48 to here be done once?  that is, does Hosting create a single `Startup` instance and therefore support use of a `bool _isAutoFac;` field?